core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -89,13 +98,25 @@ public static void writeSegmentFile(String tablePath, final String taskNo, Strin
     CarbonFile tempFolder =
         FileFactory.getCarbonFile(location + CarbonCommonConstants.FILE_SEPARATOR + tempFolderLoc);
 
-    if (tempFolder.exists() && partionNames.size() > 0) {
-      CarbonFile[] carbonFiles = tempFolder.listFiles(new CarbonFileFilter() {
-        @Override public boolean accept(CarbonFile file) {
-          return file.getName().startsWith(taskNo) && file.getName()
-              .endsWith(CarbonTablePath.INDEX_FILE_EXT);
-        }
-      });
+    if ((tempFolder.exists() && partionNames.size() > 0) || (isMergeIndexFlow
+        && partionNames.size() > 0)) {
+      CarbonFile[] carbonFiles = null;
+      if (isMergeIndexFlow) {
```

Comment:
Can we simply this logic with just flag of mergeindex outside and decide what folder to list.
Listing index& merge index both files is not a problem  for loading.